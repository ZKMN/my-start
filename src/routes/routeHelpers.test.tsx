import {
  routesConfig,
  getPublicRoutes,
  getPrivateRoutes,
} from "./routeHelpers";

describe("routeHelpers", () => {
  it("matches snapshot routesConfig", () => {
    expect(routesConfig).toMatchSnapshot();
  });

  it("testing getPublicRoutes", () => {
    const result = getPublicRoutes([{
      path: "/sessionLimit",
      component: () => <div />,
    }]);

    expect(result).toMatchSnapshot();
  });

  it("testing getPrivateRoutes", () => {
    const result = getPrivateRoutes([{
      path: "/profile",
      component: () => <div />,
    }]);

    expect(result).toMatchSnapshot();
  });
});
