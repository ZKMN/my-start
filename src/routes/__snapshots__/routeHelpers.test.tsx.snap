// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routeHelpers matches snapshot routesConfig 1`] = `
Object {
  "privateRoutes": Array [],
  "publicRoutes": Array [
    Object {
      "component": [Function],
      "path": "/",
    },
    Object {
      "component": [Function],
      "path": "/",
    },
  ],
}
`;

exports[`routeHelpers testing getPrivateRoutes 1`] = `
Array [
  <Memo(Connect(PrivateRoute))
    component={[Function]}
    exact={true}
    path="/profile"
  />,
]
`;

exports[`routeHelpers testing getPublicRoutes 1`] = `
Array [
  <Route
    component={[Function]}
    exact={true}
    path="/sessionLimit"
  />,
]
`;
